src/slice-chop.c

```diff
@@ -191,7 +191,14 @@ static SEXP chop(SEXP x, SEXP indices, struct vctrs_chop_info info) {
       ++(*info.p_index);
     }
 
-    SEXP elt = PROTECT(vec_slice_base(info.proxy_info.type, proxy, info.index));
+    // Always materialize ALTREP vectors when chopping to avoid inefficiently
```

Comment:
We should be on the lookout for potential performance improvements in the other direction (i.e. small number of large chops which eventually get discarded).